src/indicators/drought/arc2_precipitation.py

```diff
@@ -0,0 +1,710 @@
+            f"%28{date_max_dl.day}%20{date_max_dl:%b%%20%Y}%29RANGEEDGES/"
+            f"X/%28{self.range_x[0]}%29%28{self.range_x[1]}%29RANGEEDGES/"
+            f"Y/%28{self.range_y[0]}%29%28{self.range_y[1]}%29RANGEEDGES/"
+            f"data.nc"
+        )
+
+        raw_filepath = self._get_raw_filepath(master=master)
+
+        if os.path.exists(raw_filepath) and master:
+            logger.info(
+                "Removing existing master raw ARC2 file before re-downloading."
+            )
+            os.remove(raw_filepath)
+
+        cookies = {
+            "__dlauth_id": os.getenv("IRI_AUTH"),
+        }
+
+        logger.info(
+            f"Downloading ARC2 data from {self.date_min:%d %b %Y} to "
+            f"{self.date_max:%d %b %Y} "
+            f"covering longitudes {self.range_x[0]} to {self.range_x[1]} "
+            f"and latitudes {self.range_y[0]} to {self.range_y[1]}."
+        )
+
+        # TODO: explore specific errors generated by request
+        try:
+            response = requests.get(url, cookies=cookies, verify=False)
+        except requests.exceptions.RequestException as e:
+            raise SystemExit(e)
+
+        if response.status_code == 404:
+            logger.error("No data available from %s.", url)
+            return
+
+        # create folders if necessary and write file
+        Path(raw_filepath.parent).mkdir(parents=True, exist_ok=True)
+
+        with open(raw_filepath, "wb") as fd:
+            for chunk in response.iter_content(chunk_size=128):
+                fd.write(chunk)
+
+        # merge to master and delete temporary files
+        if not master:
+            master_filepath = self._get_raw_filepath(master=True)
+            if not os.path.isfile(master_filepath):
+                logger.error(
+                    (
+                        "Master raster data file not set. This needs "
+                        "to be done once by running "
+                        "`download_data(master=True)` "
+                        "for a unique set of coordinates."
+                    )
+                )
+
+            logger.info(
+                f"Merging ARC2 data from {self.date_min:%d %b %Y} "
+                f"to {self.date_max:%d %b %Y} "
+                f"into master file: {os.path.basename(master_filepath)}."
+            )
+
+            with xr.open_dataarray(master_filepath) as ds:
+                master = ds.load()
+            with xr.open_dataarray(raw_filepath) as ds:
+                raw = ds.load()
+
+            master_merge = xr.concat([master, raw], dim="T")
+
+            os.remove(raw_filepath)
+            os.remove(master_filepath)
+
+            if "_FillValue" in master_merge.encoding and np.isnan(
+                master_merge.encoding["_FillValue"]
+            ):
+                master_merge.encoding["_FillValue"] = -999
+
+            master_merge.to_netcdf(master_filepath)
+
+        return
```

Comment:
I don't know what official best practice is, but [discussions](https://stackoverflow.com/questions/13307158/most-pythonic-way-of-function-with-no-return) seem to converge to not add `return` if it never returns anything